Sources/UIScrollView.swift

```diff
@@ -77,6 +71,50 @@ open class UIScrollView: UIView {
             addSubview($0)
         }
     }
+    
+    // When adding subviews, make sure that the scroll indicators stay on top.
+    // Subview list goes back-to-front, so indicators need to take the last two positions.
+    // We use the earliestIndicatorInSubviewHierarchy variable to make sure new views are always inserted below them.
+    private lazy var earliestIndicatorInSubviewHierarchy: UIView? = {
+        let indexOfHorizontal = subviews.index(of: horizontalScrollIndicator)
+        let indexOfVertical = subviews.index(of: verticalScrollIndicator)
+
+        // if both indicators are missing, return nil, if one is missing, return one that's present
+        if indexOfVertical == nil && indexOfHorizontal == nil { return nil }
+        if indexOfHorizontal == nil { return verticalScrollIndicator }
+        if indexOfVertical == nil { return horizontalScrollIndicator }
+
+        // if both indicators are in the view hierarchy, return the one that is last
+        if indexOfHorizontal! < indexOfVertical! {
+            return horizontalScrollIndicator
+        } else {
+            return verticalScrollIndicator
+        }
+    }()
+
+
+    open override func addSubview(_ view: UIView) {
```

Comment:
I'm realizing that I should have overriden the `subviews` variable. Not only does it probably make `insertSubview` simpler, but should also just be there. Currently the behavior for `print(scrollView.subviews.count)` will differ from iOS in that it will also count the indicators.

Unfortunately, if I do override it, those versions of `addSubview` and `insertSubview` won't work as expected. I'm trying to figure it out now, noting here so all information about remaining todos is in one place.

Reply:
Update: no need to override scrollviews, iOS does have them

Reply:
@janek here you reported something different regarding subviews, or did I misunderstand one of the two pieces of info incorrectly :)